homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,205 @@
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="lyric_coordinator",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=120),
+    )
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class LyricEntity(CoordinatorEntity):
+    """Defines a base Honeywell Lyric entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        key: str,
+        name: str,
+        icon: Optional[str],
+    ) -> None:
+        """Initialize the Honeywell Lyric entity."""
+        super().__init__(coordinator)
+        self._key = key
+        self._name = name
+        self._icon = icon
+        self._location = location
+        self._mac_id = device.macID
+        self._device_name = device.name
+        self._device_model = device.deviceModel
+        self._update_thermostat = coordinator.data.update_thermostat
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return self._key
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the mdi icon of the entity."""
+        return self._icon
+
+    @property
+    def location(self) -> LyricLocation:
+        """Get the Lyric Location."""
+        return self.coordinator.data.locations_dict[self._location.locationID]
+
+    @property
+    def device(self) -> LyricDevice:
+        """Get the Lyric Device."""
+        return self.location.devices_dict[self._mac_id]
+
+
+class LyricDeviceEntity(LyricEntity):
+    """Defines a Honeywell Lyric device entity."""
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information about this Honeywell Lyric instance."""
+        return {
+            "connections": {(dr.CONNECTION_NETWORK_MAC, self._mac_id)},
+            "manufacturer": "Honeywell",
+            "model": self._device_name,
+            "name": self._device_model,
```

Comment:
Does the device provide a `"swversion"` ?

Reply:
On the device, yes. On the API, nope, no idea why...

I can read the firmware and bunch of other versions on the thermostat itself, but the API doesn't expose this info